crates/uv-python/src/downloads.rs

```diff
@@ -415,6 +415,72 @@ impl PythonDownloadRequest {
         self.version.take()
     }
 
+    /// Remove default implementation and platform details so the request only contains
+    /// explicitly user-specified segments.
+    #[must_use]
+    pub fn unset_defaults(self) -> Self {
+        self.unset_defaults_with_host(Platform::from_env().ok().as_ref())
+    }
+
+    pub(crate) fn unset_defaults_with_host(mut self, host: Option<&Platform>) -> Self {
```

Comment:
It seems weird to allow `Option` here?

I think I'd have `unset_defaults` include the implementation filtering, then have `unset_platform_defaults` include the platform specific part and do the `let Some(host) ...` matching in `unset_defaults`. Does that make sense?